tests/trainer/test_trainer.py

```diff
@@ -1042,7 +1042,7 @@ def configure_gradient_clipping(self, *args, **kwargs):
             # test that gradient is clipped correctly
             parameters = self.parameters()
             grad_norm = torch.norm(torch.stack([torch.norm(p.grad.detach(), 2) for p in parameters]), 2)
-            torch.testing.assert_allclose(grad_norm, torch.tensor(0.05))
+            torch.testing.assert_allclose(grad_norm, torch.tensor(0.05).type_as(grad_norm))
```

Comment:
Why is this necessary?

Reply:
Otherwise, it will complain the tensor isn't on the device...
```
AssertionError: The values for attribute 'device' do not match: cuda:0 != cpu.
```

Reply:
But why `type_as` and not `to(grad_norm.device)`?

I find it unintuitive that `type_as` moves also the device

Reply:
Ah nvm you updated it already